tabs/tab_survival.py

```
@@ -1,511 +1,812 @@
+    def current_df() -> Optional[pd.DataFrame]:
+        """Select between original and matched dataset based on user preference."""
+        if is_matched.get() and input.radio_survival_source() == "matched":
+            return df_matched.get()
+        return df.get()
     
-    surv_df_current = reactive.Value(None)  # Current dataset (original or matched)
-    curves_result = reactive.Value(None)    # KM/NA analysis result
-    landmark_result = reactive.Value(None)  # Landmark analysis result
-    cox_result = reactive.Value(None)       # Cox regression result
+    @render.ui
+    def ui_title_with_summary():
```

Ruff (0.14.10)

Missing return type annotation for private function `ui_title_with_summary`

(ANN202)